### ISSUE TITLE
autocmd is adding duplicate match rules

### ISSUE THREAD

**Issue by dpatti** [chr4/sslsecure.vim#5]
This plugin is leaking matches into the vim match list, causing it to grow and greatly slow things down over time with the overwhelming number of matches.

Without the plugin:

```
:echo len(getmatches())
60
:e file
"file" [New File]
:e file
"file" [New File]
:e file
"file" [New File]
:echo len(getmatches())
60
:e file
"file" [New File]
:e file
"file" [New File]
:e file
"file" [New File]
:echo len(getmatches())
60
```

With the plugin:

```
:echo len(getmatches())
239
:e file
"file" [New File]
:e file
"file" [New File]
:e file
"file" [New File]
:echo len(getmatches())
776
:e file
"file" [New File]
:e file
"file" [New File]
:e file
"file" [New File]
:echo len(getmatches())
1313
```

**Comment by chr4** [comment]
Thanks for reporting this! Performance should not be harmed beyond necessary. Do you happen to have an idea howto fix this? Would an autocmd group help?

**Comment by dpatti** [comment]
I can't say that I know for sure how you're supposed to add matches. If it is a per-window setting, then you might have more success with `WinNew` instead of `BufWinEnter` as far as the leak goes, but it still seems like there's a better way.

**Comment by frioux** [comment]
Yeah, you should add autocommands like this:

``` vim
augroup sslsecure
  autocmd!

  autocmd BufWinEnter * call s:genmatch('SSLv3', '', '')
  autocmd BufWinEnter * call s:genmatch('SSLv2', '', '')
  autocmd BufWinEnter * call s:genmatch('HIGH', '', '')
  " ...
augroup END
```

The above will ensure that each is only in use exactly once.

**Comment by chr4** [comment]
Thanks for the hint!
I've pushed a [commit](https://github.com/chr4/sslsecure.vim/commit/7114323a330249e330bb2b7a456cd957b6884677) that wraps all `autocmd` statements into an `augroup`. This is probably a good idea in general, but it seems to not be enough to solve the issue @dpatti pointed out...

**Comment by chr4** [comment]
Released in `v1.1.1`
